### PR TITLE
Fix class name when generating metrics for Object Detection

### LINKED ISSUE #414: MscocoObjectDetection does not work.
MscocoObjectDetection does not work. On the other hand, Similar dataloader MscocoObjectDetectionPerson works.
This is the trace back:
```
Traceback (most recent call last):
  File "executor/train.py", line 432, in <module>
    main()
  File "/storage/fukasawa/.pyenv/versions/benchmark_mscoco/lib/python3.6/site-packages/click/core.py", line 722, in __call__
    return self.main(*args, **kwargs)
  File "/storage/fukasawa/.pyenv/versions/benchmark_mscoco/lib/python3.6/site-packages/click/core.py", line 697, in main
    rv = self.invoke(ctx)
  File "/storage/fukasawa/.pyenv/versions/benchmark_mscoco/lib/python3.6/site-packages/click/core.py", line 895, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/storage/fukasawa/.pyenv/versions/benchmark_mscoco/lib/python3.6/site-packages/click/core.py", line 535, in invoke
    return callback(*args, **kwargs)
  File "executor/train.py", line 428, in main
    run(network, dataset, config_file, experiment_id, recreate)
  File "executor/train.py", line 393, in run
    start_training(config)
  File "executor/train.py", line 136, in start_training
    metrics_summary_op, metrics_placeholders = executor.prepare_metrics(metrics_ops_dict)
  File "/storage/fukasawa/tmp/blueoil_mscoco/lmnet/lmnet/utils/executor.py", line 121, in prepare_metrics
    tf.float32, name="{}_placeholder".format(metrics_key)
  File "/storage/fukasawa/.pyenv/versions/benchmark_mscoco/lib/python3.6/site-packages/tensorflow/python/ops/array_ops.py", line 2077, in placeholder
    return gen_array_ops.placeholder(dtype=dtype, shape=shape, name=name)
  File "/storage/fukasawa/.pyenv/versions/benchmark_mscoco/lib/python3.6/site-packages/tensorflow/python/ops/gen_array_ops.py", line 5791, in placeholder
    "Placeholder", dtype=dtype, shape=shape, name=name)
  File "/storage/fukasawa/.pyenv/versions/benchmark_mscoco/lib/python3.6/site-packages/tensorflow/python/framework/op_def_library.py", line 394, in _apply_op_helper
    with g.as_default(), ops.name_scope(name) as scope:
  File "/storage/fukasawa/.pyenv/versions/benchmark_mscoco/lib/python3.6/site-packages/tensorflow/python/framework/ops.py", line 6088, in __enter__
    return self._name_scope.__enter__()
  File "/storage/fukasawa/.pyenv/versions/3.6.4/lib/python3.6/contextlib.py", line 81, in __enter__
    return next(self.gen)
  File "/storage/fukasawa/.pyenv/versions/benchmark_mscoco/lib/python3.6/site-packages/tensorflow/python/framework/ops.py", line 3989, in name_scope
    raise ValueError("'%s' is not a valid scope name" % name)
ValueError: 'AveragePrecision_0.5/traffic light_placeholder' is not a valid scope name
```

It seems that an error has occurred because there is a space in the class name such as ‘traffic light’. I think that it is necessary to add processing that replaces spaces with underscores or something.

### PR DESCRIPTION
**Pull request by Joeper214**
## Motivation and Context

The class name cannot be generated from tensorboard metrics this PR fixes this Issue.
- Fix #414 

## How has this been tested?

- Try to train an yolov2 or yolov1 task using MScoco dataset. 

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature / Optimization (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: lmnet/lmnet/networks/object_detection/yolo_v1.py [hunk 1]
```diff
@@ -98,6 +98,7 @@ def metrics(self, output, labels, thresholds=[0.3, 0.5, 0.7]):
         for overlap_thresh in thresholds:
             average_precisions = []
             for class_id, class_name in enumerate(self.classes):
+                class_name = class_name.replace(' ', '_')
```
Comment (iizukak) [inline lmnet/lmnet/networks/object_detection/yolo_v1.py]: @Joeper214 
Thank you for fixing!!

One question.
In you code, same changes exist in two networks.
It's not possible to fix in one change? (maybe in base class?)

Comment (Joeper214) [reply lmnet/lmnet/networks/object_detection/yolo_v1.py]: Yeah. That's a better idea. I was thinking about using Lambda function on base class. to filter or replace spaces with an underscore.

Comment (Joeper214) [reply lmnet/lmnet/networks/object_detection/yolo_v1.py]: @iizukak Updated in the next commit thanks!

File: lmnet/lmnet/networks/base.py [hunk 2]
```diff
@@ -58,7 +58,7 @@ def __init__(
         self.optimizer_kwargs = optimizer_kwargs if optimizer_kwargs is not None else {'learning_rate': 0.01}
         self.learning_rate_func = learning_rate_func
         self.learning_rate_kwargs = learning_rate_kwargs if learning_rate_kwargs is not None else {}
-        self.classes = classes
+        self.classes = list(map(lambda _class: _class.replace(' ', '_'), classes))
```
Comment (iizukak) [inline lmnet/lmnet/networks/base.py]: @Joeper214 
Why convert with `list` is needed?

Comment (Joeper214) [reply lmnet/lmnet/networks/base.py]: Oh. I was thinking of parsing the `map` object back to `list`. but it seems not necessary thanks for pointing out.

Comment (Joeper214) [reply lmnet/lmnet/networks/base.py]: Updated in next commit.

Comment (Joeper214) [reply lmnet/lmnet/networks/base.py]: I checked the test case it seems that test fails if we don't parse the `map` object back to `list`.
```

[2019-10-08T01:54:22Z]         self.classes = classes
--
  | [2019-10-08T01:54:22Z] >       self.num_classes = len(classes)
  | [2019-10-08T01:54:22Z] E       TypeError: object of type 'map' has no len()
  | [2019-10-08T01:54:22Z]
  | [2019-10-08T01:54:22Z] lmnet/networks/object_detection/yolo_v2.py:1044: TypeError
```

Comment (iizukak) [reply lmnet/lmnet/networks/base.py]: I see.

**Comment by tkng** [comment]
Readability Approval

**Comment by Joeper214** [comment]
@tkng @iizukak Is this okay to merge?

**Comment by tkng** [comment]
Yes, I just waited the completion of unit tests.

**Pull request merged**